Spice up Enchanted Armoire messages fixes #9233

spacenate opened the pull request:
Fixes #9233

This PR is not yet complete, I am seeking feedback/clarification first

### Changes
I have updated the "text" for notifications that are created when a user purchases an armoire item. I am not sure this is the correct notification mechanism to be editing however, due to some style issues I ran in to while doing this. Note the overlapping text and image in the first screenshot, and poor layout in second screenshot.

![screen shot 2017-10-20 at 9 09 32 am](https://user-images.githubusercontent.com/11892841/31828749-7271bacc-b578-11e7-8fc1-13c501d58e3e.png)

![screen shot 2017-10-20 at 9 09 21 am](https://user-images.githubusercontent.com/11892841/31828758-7671c4aa-b578-11e7-8d88-d1f15f5315e8.png)

Am I editing the right notification text, and should I fix the styles in addition to editing the text? Or is there a different notification/popup mechanism that would be better to use? Thanks!

This PR also includes a second, unrelated change: I fixed a misspelled i18n translation key. I verified it was a misspelling by searching the codebase for the misspelled version (1 result), and then the properly spelled version (many results).

----
UUID: 0de0861f-0491-41a1-899b-43488e0cf8b5

spacenate committed:
Updates notification text for armoire items in shops:genericPurchase.
Adjusts column widths for 'drop' type notifications to prevent .text from overlapping .icon-item.
Adds new 'flavorMessage' property to Snackbar notifications, which overrides standard computed message.
Vertically center XP gain icon/amount.
Fixes unrelated typo in i18n key name.

lemoness commented:
@tressley You'll need to weigh in on the design stuff in this PR!

spacenate commented:
I apologize for botching the git history a bit. 5d55941 contains a few more changes:
-  Show experience gains with two separate notifications, as suggested by @citrusella. I used the `success` notification type, but `info` also seemed to work well in this situation.
- Adjust column widths for 'drop' type notifications to prevent text from overlapping icons. It seems the existing column widths (2 and 9) were not well suited to the 64px wide `.icon-item` element.

![screen shot 2017-10-20 at 10 49 28 pm](https://user-images.githubusercontent.com/11892841/31848186-6856cba6-b5e9-11e7-8602-b83df0b12470.png)
![screen shot 2017-10-20 at 10 49 43 pm](https://user-images.githubusercontent.com/11892841/31848187-686ccaa0-b5e9-11e7-816a-fe975a883249.png)

Thoughts? 🌸

spacenate commented:
🤦‍♂️ My previous transgression (rebasing after pushing) is still haunting me in this branch.

fc09ea4 updates Enchanted Armoire to only use 1 notification for XP gains, per @Tressley.

Screenshot showing two separate purchases (one drop, one XP)

![screen shot 2017-10-23 at 2 09 16 pm](https://user-images.githubusercontent.com/11892841/31912271-913763fa-b800-11e7-9a32-a12f547168ec.png)

To alter the standard computed message, I introduced a new property to for Snackbar notification objects, `flavorMessage`. 

Meow?

Tressley commented:
Hey @spacenate -- I'd like to invite you to Habitica's Zeplin project. We use Zeplin to manage our mock-up specs for our developers and designers. Is there an email address you wouldn't mind sharing for an invite? I think getting you access to our Notifications specs will definitely help out!

SabreCat commented:
I'm liking the direction this is headed! Something's not right with the alignment of things, though, with the equipment icon flying into the upper left corner and the Experience star floating to the top. (If that's a preexisting issue, though, it may not be necessary to fix in this PR's scope.)

spacenate commented:
@Tressley Sure, you may use nathan@spacenate.com

spacenate commented:
I rebased my changes to clean up the git history, is now easier to see what has been changed in this PR 👍

spacenate commented:
@SabreCat I know it was consistent at the time, but now after rebasing, equipment icons no longer fly to the upper left corner 🤷‍♂️  The experience star rising to the top seems to be pre-existing, and due to variable height of message/flavorMessage I didn't see a simple solution for centering it vertically (table cell hack, or flexbox maybe?).

I'll await a Zeplin invitation, or let me know if there are any other changes to be made to this PR in the meantime.

spacenate commented on website/common/script/ops/buyArmoire.js:
```diff
@@ -42,11 +42,12 @@ module.exports = function buyArmoire (user, req = {}, analytics) {
 
     user.items.gear.owned[drop.key] = true;
     user.flags.armoireOpened = true;
+    // @nate -- already using correct i18n strings in this file
```
Oops, I'll remove this comment

Tressley commented:
@spacenate -- Zeplin invite sent your way. Sorry for the delay!

spacenate commented:
@Tressley For some reason, I never received the Zeplin invite (I did check in my junk folder) 🤷‍♂️ May I ask if there any changes that should be made in this PR for #9233?

![screen shot 2017-10-27 at 1 24 44 pm](https://user-images.githubusercontent.com/11892841/32249527-f552f62a-be4e-11e7-85db-c48b3f328724.png)
![screen shot 2017-10-27 at 1 25 27 pm](https://user-images.githubusercontent.com/11892841/32249528-f5656fb2-be4e-11e7-9385-938484c96c76.png)

Tressley commented:
@spacenate -- Thanks for the heads up. I just resent the invite, please let me know if it doesn't show up.

Regarding the changes, is it possible to vertically center the Experience icon and value in addition to prepending a "+" to dictate gaining EXP? Thanks!

paglias commented:
hey @spacenate how are you doing on this?

spacenate commented:
Hello, sorry for my absence. @Tressley, I never did get that Zeplin invite.

I have rebased this PR with the most recent from upstream/develop, and added the two changes requested (vertically center the Experience icon and value in addition to prepending a "+" to dictate gaining EXP).

![screen shot 2018-01-02 at 6 10 09 pm](https://user-images.githubusercontent.com/11892841/34506304-cec2f4c6-efe8-11e7-9c86-be321b4390ba.png)

Anything else needed here? Sorry again for the delays 💩

Tressley commented:
That's odd @spacenate. I've sent invites twice now and it is still showing as pending. Can you check your spam folder?
![image](https://user-images.githubusercontent.com/2686586/34506382-32f1202e-effa-11e7-805e-a7fc358f333a.png)

spacenate commented:
Yay, I finally got one of the invites! Just after commenting here yesterday. I am now in Zeplin.

spacenate commented:
There do not appear to be any screens related to this PR in Zeplin. Are there any other changes this PR needs to fix #9233?

Tressley commented:
@spacenate -- Sorry about that. I've uploaded the necessary art board to Zeplin here: https://zpl.io/a8ypMxE

This should give you a better idea of how these Snacks should be styled.

spacenate commented:
The new styles seem to be addressed in #9648. If @marcusvmdomingos does not submit a PR for that issue, I may decide to take it on. Is there anything else missing from my PR to fix #9233?

marcusvmdomingos commented:
Hey, @spacenate . I started to work on this yesterday. It's my first contribution to Habitica, so I had some issues to prepare dev enviroment. Anyway, it's kinda working, but I'm getting an error while trying to complete a task to test the styles of notifications.

If you are "close" to resolve the issue related to #9648 you can get it, since I probably will spend more time getting it to work properly before make the changes to fix the problem. I will do it even if you take the issue (I don't mind, really), just for gaining experience... then I can take other issues to do later.

So, if you want it, just notify us in #9648 for proper organization and assignment. Thank you!

spacenate commented:
Hi @marcusvmdomingos, I have not done any work related to #9648. In this PR I addressed #9233, and made a minor style adjustment as requested by @Tressley.

marcusvmdomingos commented:
So let me try to do it. In a week I may have it done.

2018-01-04 21:59 GMT-02:00 Nathan Sollenberger <notifications@github.com>:

> Hi @marcusvmdomingos <https://github.com/marcusvmdomingos>, I have not
> done any work related to #9648
> <https://github.com/HabitRPG/habitica/issues/9648>. In this PR I
> addressed #9233 <https://github.com/HabitRPG/habitica/issues/9233>, and
> made a minor style adjustment as requested by @Tressley
> <https://github.com/tressley>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitica/pull/9240#issuecomment-355434240>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Abghi443bGi-ufqGpoSxoe-LLjwPkSwXks5tHWXVgaJpZM4QA1iW>
> .
>



-- 
Qualquer dúvida, estou à disposição.

att

TheHollidayInn commented:
@spacenate Your title is marked WIP. Is this PR ready for review or do you still need time or need answers from us?

TheHollidayInn commented:
This is working well locally!

spacenate commented:
Doh! No, this should be complete now. I will remove the WIP bit :)

paglias commented:
@spacenate thanks for this! I'm merging the PR and awarding your contributor tier.

Note that further tiers require increasing amounts of work from one to the next, so Tier 2 may require a couple of PRs or a larger PR to attain. But keep helping out and we'll express our gratitude accordingly!
